lib/SILGen/SILGenConvert.cpp

```diff
@@ -613,12 +613,21 @@ ManagedValue SILGenFunction::emitExistentialErasure(
                             const TypeLowering &existentialTL,
                             ArrayRef<ProtocolConformanceRef> conformances,
                             SGFContext C,
-                            llvm::function_ref<ManagedValue (SGFContext)> F,
+                            function_ref<ManagedValue (SGFContext)> inputF,
                             bool allowEmbeddedNSError) {
   // Mark the needed conformances as used.
   for (auto conformance : conformances)
     SGM.useConformance(conformance);
 
+  // A wrapper around inputF that makes sure we are passed a +1 value if we are
```

Comment:
This feels like a hack. Why not convert it to a +1 value where necessary when you actually call F?

Reply:
Actually isn't the +1-ness supposed to be encoded in the SGFContext? The original intent of the code was that everything produced a +1 value unless SGFContext.allowPlusZero() was true.

Reply:
@slavapestov I don't think it makes sense to do existential erasure on +0 value. You are producing a new value independent of the original value. That says it must be an owned value.

Reply:
I am converting it to the +1 when calling F in the caller. This is just the callee can assert that is true in a way that should optimize out in no-asserts build.

I am fine changing the code in the caller to check the SGFContext though instead of doing ensurePlusOne.